### PR TITLE
[#9986] Allow to disable fetching of tx history on mobile network

### LINKED ISSUE #9986: Allow to disable fetching of tx history on mobile network
# Allow to disable fetching of tx history on mobile network

[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a user, I want to have an option to disable fetching of tx history on mobile network to save mobile traffic. 

### Description

Currently we fetch history regardless of available network, and this might require some internet traffic (tens of megabytes on initial scan if account contains transactions and hundreds of megabytes if when listening to recent history during the day). This can be expansive for some users.

### Acceptance Criteria

[comment]: # (Rules for the future PR to be accepted.)

### Notes

[comment]: # (For the developer who will implement. Include planning poker estimates, guidance, etc.)

### PR DESCRIPTION
**Pull request by rasom**
The esiest way to test it would be to compare data usage on release version and this one. You would need to switch to mobile network, select "stop syncing", and then keep app in the foreground for 5mins. 

fixes #9986

status: ready

### CODE REVIEW

File: src/status_im/ui/screens/wallet/accounts/views.cljs [hunk 1]
```diff
@@ -180,6 +182,39 @@
...
 
+(defn wallet-service-state []
+  (let [status  @(re-frame/subscribe [:network-status])
+        type    @(re-frame/subscribe [:network/type])
+        setting @(re-frame/subscribe [:wallet-service/manual-setting])
+        state   @(re-frame/subscribe [:wallet-service/state])]
+    [react/view
+     {:style {:flex 1
+              :flex-direction :row}}
+     [react/text {:style {:flex 1}} (str "setting: " setting)]
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/accounts/views.cljs]: i18n?

Comment (flexsurfer) [reply src/status_im/ui/screens/wallet/accounts/views.cljs]: or is this for testing?

File: src/status_im/wallet/core.cljs [hunk 2]
```diff
@@ -601,3 +602,77 @@
...
+                "state"    state
+                "network-type" network-type
+                "network-status" network-status)
+      (when (and config/disable-wallet-on-mobile-network?
+                 (or
+                  (nil? network-status)
+                  (= network-status :online))
+                 (nil? setting))
+        (cond
+          (and (not= network-type "wifi")
```
Comment (flexsurfer) [inline src/status_im/wallet/core.cljs]: constant?

File: src/status_im/ui/screens/wallet/accounts/views.cljs (hunk 1 above)
Comment (rasom) [reply src/status_im/ui/screens/wallet/accounts/views.cljs]: yep, that's temporary

**Review by flexsurfer — APPROVED** [review]

**Comment by hesterbruikman** [comment]
@rasom This is a good solution. Didn't learn too much about bandwidth use in target markets unfortunately, but anyway we can reduce and give control makes sense. Does `Force start` enable syncing over mobile data?

@errorists what are your thoughts on Enable/Disable message syncing and tx syncing over mobile data? Should this be a single setting or should both be controlled separately? As this impacts Settings and the Sync over Mobile bottom sheet

**Comment by errorists** [comment]
@hesterbruikman it should be a single setting and unified with the chat dialogues

**Comment by rasom** [comment]
@hesterbruikman @errorists 
Done, fetching of tx history is disabled/enabled on mobile with the same setting as message history.

**Comment by hesterbruikman** [comment]
🙌 great! Thanks @rasom 

I think this needs minor copy changes to accurately represent the mobile data switch. What do you think of below @errorists @rasom This is the fastest route I can think of to include in this PR. You might want to better organize the sync settings though. If so I'll create a separate issue

_Note that white line in the bottom sheet is weird_ 🤔

Frame 1

**Comment by rasom** [comment]
@hesterbruikman
translations are fixed (though EN only)


> You might want to better organize the sync settings though. If so I'll create a separate issue
 
We could have two options there: 

1) keep syncing both messages and transactions
2) keep syncing only messages

As far as I understand currently wallet requires much more traffic than messages (@cammellos @oskarth do you confirm?). There are many users who use the app as a messenger in the first place, not as a wallet. For them it might make sense to keep syncing only messages history. And we can also add a button in wallet which will refresh history on demand without listening to the network (thus without using mobile data constantly).

**Comment by hesterbruikman** [comment]
> And we can also add a button in wallet which will refresh history on demand without listening to the network (thus without using mobile data constantly).

If you're feeling adventurous, this is a long outstanding issue to reduce frequency of fetching and include fetching on demand: https://github.com/status-im/status-react/issues/10242 (pull to refresh). Having this manual fetching, I think we could keep the mobile data setting a catch all `mobile sync` for messaging and tx

Mobile data
- Enable - _auto-sync messages and tx when on mobile network, pull to refresh available in addition_
- Disable - does not sync messages and tx prompt settings bottom sheet when:_
   - _switching to mobile network_
   - _attempting pull to refresh_
- Data saver - _syncs tx on pull to refresh only_


Potential addition to the original design of pull to refresh could be:
If user pulls to refresh while on mobile data, UI could prompt the `Sync using mobile data?` currently used for messaging

**Comment by rasom** [comment]
@hesterbruikman I would merge this PR at first and then take a look at #10242  later

**Comment by hesterbruikman** [comment]
Sounds good to me. Also without #10242 being able to switch off sync over mobile data for messages _and_ tx is a win if you want to spare your data plan. Let's consider `datasaver` as a separate feature

**Comment by churik** [comment]
@rasom 
### ISSUE / QUESTION 1 IOS vs Android - different behavior 
**Steps:** 
1) turn off WI-FI (LTE on IOS, 4G on Android)
2) restore account with transaction history

**On IOS(LTE)**: 
- initial scanning: transaction history is not loaded, usage: 1,8 MB on current PR build vs 1,7 MB on release build
- foreground for 5 mins: 0,4 MB on current PR build vs 11,3 MB on release build

**on Android(4G+)**:
- initial scanning: transaction history is loaded very quickly, usage: 12,9 MB on current PR build VS  13,2 on release build
- foreground for 5 mins: 7.1 MB on current PR build vs 5,3 MB on release build
So not sure that feature is working on Android (note: automatic lock is turned on on Android, so can be that app was on background for a couple of moments)

Logs from PR build, Android: [Status-debug-logs.zip](https://github.com/status-im/status-react/files/5063311/Status-debug-logs.zip)



### ISSUE 2 On IOS when creating account on mobile network after enabling wifi transaction history is not loaded until app restart (also in nightly)

As it is an issue in nightly as well, can be reported separately, so it is in case you can take a look in this PR into this.
**Steps:** 
1) turn off WI-FI (LTE on IOS)
2) restore account with transaction history
3) enable wi-fi

**Expected result:** transaction history is loaded
**Actual result:** transaction history is not  loaded until app restart
**Log:** [status_logs.zip](https://github.com/status-im/status-react/files/5063404/status_logs.zip)

**Comment by rasom** [comment]
> ISSUE 2 On IOS 

is issue 2 iOS specific?

**Comment by churik** [comment]
It is not reproducible on Android, only on IOS. Tried on IOS 13.5.1, however, can try on another IOS versions if it makes sense

**Comment by rasom** [comment]
@churik both issues should be fixed now

**Comment by churik** [comment]
**Android:** 
Initial scanning: 158 KB on current PR build VS 13,2 MB on release build
Foreground 5 min (with coming back to background and back): 409 KB (2 messages in 1-1 chat were received) on current PR build vs 5,3 MB on release build
**IOS:**
Initial scanning: 150 KB on  current PR build VS 1,7 MB on release build
Foreground 5 min (with coming back to background and back):  100 KB vs 11,3 MB on release build

Messaging in 1-1 chats is still working while on 3G network.
Checked, that transactions are downloaded when turning in  WI-Fi or start syncing and disabled again when turning it off.

GREAT WORK!

**Pull request merged**